app/controllers/accept_account_invitations_controller.rb

```diff
@@ -1,18 +1,24 @@
 # frozen_string_literal: true
 
 class AcceptAccountInvitationsController < ApplicationController
-  before_action :authenticate_user!
+  def show
+    return if user_signed_in? && the_invitation
 
-  def show; end
+    session[:token] = ps[:token]
```

Comment:
Better remember `after_sign_in_url` in the session, this way the after-sign-in redirect implementation will be easy without knowing the internals of possible workflows current or future.

Reply:
Global. Beautiful.